src/Features/CSharp/Portable/QualifyMemberAccess/CSharpQualifyMemberAccessDiagnosticAnalyzer.cs

```diff
@@ -15,7 +17,21 @@ protected override bool IsAlreadyQualifiedMemberAccess(SyntaxNode node)
             => node.IsKind(SyntaxKind.ThisExpression);
 
         // If the member is already qualified with `base.`, it cannot be further qualified.
-        protected override bool CanMemberAccessBeQualified(SyntaxNode node)
-            => !node.IsKind(SyntaxKind.BaseExpression);
+        protected override bool CanMemberAccessBeQualified(ISymbol containingSymbol, SyntaxNode node)
+            => !(node.IsKind(SyntaxKind.BaseExpression) || IsInPropertyOrFieldInitialization(containingSymbol, node));
+
+        private bool IsInPropertyOrFieldInitialization(ISymbol containingSymbol, SyntaxNode node)
```

Comment:
💭 I'm a bit concerned about the performance implications of this. However, when I went to investigate how [`ThisKeywordRecommender`](https://github.com/dotnet/roslyn/blob/master/src/Features/CSharp/Portable/Completion/KeywordRecommenders/ThisKeywordRecommender.cs) implemented the same feature, I found a bug. I'll file a bug to find a better way to implement this check and unify this analyzer and `ThisKeywordRecommender` to use it.